### PR TITLE
[bitnami/drupal] (feat) Allows users to optionally install Drupal dependencies with Composer

### PR DESCRIPTION
**Pull request by scottrigby**
### Description of the change

Setting `DRUPAL_COMPOSER_INSTALL` env var to `yes` will install Drupal dependencies with Composer.

If `DRUPAL_COMPOSER_INSTALL` is `yes`, the following additional env vars apply:
* `DRUPAL_COMPOSER_FILE` allows users to specify an alternative desired Drupal Composer file location. This is useful for some setups, where you want composer.json to be a directory above the Drupal installation dir.
* `DRUPAL_MOUNTED_COMPOSER_FILE` Location of a mounted custom Composer file
* `DRUPAL_MOUNTED_COMPOSER_LOCK_FILE`: Location of a mounted custom Composer lock file. A lock file is recommended in testing and production environments for dependency version reproducibility. This warning is presented to the user if a Composer lock file is not also mounted.

Example custom composer.json file change to add Token contrib module :

```diff
@@ -21,7 +21,8 @@
         "drupal/core-recommended": "^9.5",
         "drupal/core-vendor-hardening": "^9.5",
         "drush/drush": "11.5.1",
-        "phpmailer/phpmailer": "6.8.0"
+        "phpmailer/phpmailer": "6.8.0",
+        "drupal/token": "^1.11"
     },
     "conflict": {
         "drupal/drupal": "*"
```

Example using Docker compose:

```diff
--- a/bitnami/drupal/10/debian-11/docker-compose.yml
+++ b/bitnami/drupal/10/debian-11/docker-compose.yml
@@ -21,8 +23,13 @@ services:
       - DRUPAL_DATABASE_NAME=bitnami_drupal
       # ALLOW_EMPTY_PASSWORD is recommended only for development.
       - ALLOW_EMPTY_PASSWORD=yes
+      - DRUPAL_COMPOSER_INSTALL=true
+      - DRUPAL_MOUNTED_COMPOSER_FILE=/custom/mounted-composer.json
+      - DRUPAL_MOUNTED_COMPOSER_LOCK_FILE=/custom/mounted-composer.lock
     volumes:
       - 'drupal_data:/bitnami/drupal'
+      - './test-data/test-mounted-composer.json:/custom/mounted-composer.json'
+      - './test-data/test-mounted-composer.lock:/custom/mounted-composer.lock'
     depends_on:
       - mariadb
 volumes:
```

### Benefits

This is not an edge case. Composer is the preferred way to install Drupal dependencies, such as contrib or custom modules, themes, and install profiles, patches to any of those or to Drupal core. However, there is currently no good way to do this with the bitnami/drupal container for initial installation (see https://www.drupal.org/docs/develop/using-composer/manage-dependencies).

### Possible drawbacks

None.

### Applicable issues

- fixes #

N/A

### CODE REVIEW

File: bitnami/drupal/10/debian-11/rootfs/opt/bitnami/scripts/drupal-env.sh [hunk 1]
```diff
@@ -69,6 +69,11 @@ unset drupal_env_vars
 export DRUPAL_BASE_DIR="${BITNAMI_ROOT_DIR}/drupal"
 export DRUPAL_CONF_FILE="${DRUPAL_BASE_DIR}/sites/default/settings.php"
 export DRUPAL_MODULES_DIR="${DRUPAL_BASE_DIR}/modules"
+export DRUPAL_COMPOSER_INSTALL="${DRUPAL_COMPOSER_INSTALL:-no}"
+default_drupal_composer_file="${DRUPAL_BASE_DIR}/composer.json"
+export DRUPAL_COMPOSER_FILE="${DRUPAL_COMPOSER_FILE:-$default_drupal_composer_file}"
```
Comment (rafariossaa) [inline bitnami/drupal/10/debian-11/rootfs/opt/bitnami/scripts/drupal-env.sh:74]: There is not need for an intermediate variable, you could use:

```
export DRUPAL_COMPOSER_FILE="${DRUPAL_COMPOSER_FILE:-${DRUPAL_BASE_DIR}/composer.json}"
```

File: bitnami/drupal/10/debian-11/rootfs/opt/bitnami/scripts/libdrupal.sh [hunk 2]
```diff
@@ -127,6 +127,11 @@ drupal_initialize() {
         fi
     fi
 
+    # Check if we should install Drupal dependencies with Composer
+    if is_boolean_yes "${DRUPAL_COMPOSER_INSTALL:-}"; then
+        drupal_composer_install
+    fi
+
```
Comment (rafariossaa) [inline bitnami/drupal/10/debian-11/rootfs/opt/bitnami/scripts/libdrupal.sh:134]: this could be changed into:
```
is_boolean_yes "$DRUPAL_COMPOSER_INSTALL" && drupal_compose_install
```

and `DRUPAL_COMPOSER_INSTALL` (and related env vars) values should be checked in the `drupal_validate` function.

File: bitnami/drupal/10/debian-11/rootfs/opt/bitnami/scripts/libdrupal.sh [hunk 3]
```diff
@@ -189,6 +194,55 @@ drupal_initialize() {
     true
 }
 
+########################
+# Installs dependencies with Composer
+########################
+drupal_composer_install() {
+    composer_file_dir="$(dirname "${DRUPAL_COMPOSER_FILE:-}")"
+    composer_lock_file="${composer_file_dir}/composer.lock"
```
Comment (rafariossaa) [inline bitnami/drupal/10/debian-11/rootfs/opt/bitnami/scripts/libdrupal.sh:202]: I think `DRUPAL_MOUNTED_COMPOSER_LOCK_FILE` should be used here, or am I wrong ?

**Review by rafariossaa — CHANGES_REQUESTED** [review]
Hi,
  Thanks you very much for adding this.
  Could you take a look to my comments ?
